HyperPlay-Gaming/hyperplay-desktop-client
# [FEAT] Add support for Actions when navigating to GamePage

This is a small but useful change that should fix some UX issues.

It basically adds the ability to pass an action on the location state when navigating to the game page from anywhere, including the store.

The actions can be anything but so far I limited to `install`, `launch` and `uninstall`.

The api that navigates from the store was also update with that so we can either launch or install a game with one click instead of two.

Same for the download manager, the Play icon should call the Play method and the Download icon should show the Install dialog.
If the play icon was clicked but the game is not installed then it will also show the install dialog.

Merge this one before this one: https://github.com/HyperPlay-Gaming/hyperplay-store/pull/217

### How to test

- Launch/Add any game from the store.
- If the game is installed, it will launch it.
- If the game is not installed, will open the install dialog.
- From the Download manager screen
- The play button should launch the game if installed.
- The download button should open the install dialog.

---

Use the following Checklist if you have changed something on the Backend or Frontend:
